diff --git a/arch/arm/boot/dts/am335x-bone.dts b/arch/arm/boot/dts/am335x-bone.dts
index 6b849372..f2d0402 100644
--- a/arch/arm/boot/dts/am335x-bone.dts
+++ b/arch/arm/boot/dts/am335x-bone.dts
@@ -7,6 +7,8 @@
  */
 /dts-v1/;
 
+#include <dt-bindings/board/am335x-bbw-bbb-base.h>
+
 #include "am33xx.dtsi"
 #include "am335x-bone-common.dtsi"
 
@@ -24,3 +26,32 @@
 &mmc1 {
 	vmmc-supply = <&ldo3_reg>;
 };
+
+&spi0 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&spi0_pins>;
+
+    status = "okay";
+
+    spidev@0 {
+        compatible = "spidev";
+        spi-max-frequency = <48000000>;
+        reg = <0>;
+    };
+};
+
+&am33xx_pinmux {
+	spi0_pins: pinmux_spi0 {
+		pinctrl-single,pins = <
+            /* SPI0 pins for nRF24L01+ */
+            BONE_P9_22 (PIN_INPUT_PULLUP | MUX_MODE0)  /* spi0_sclk.spi0_sclk */
+            BONE_P9_21 (PIN_INPUT_PULLUP | MUX_MODE0)  /* spi0_d0.spi0_d0 */
+            BONE_P9_18 (PIN_OUTPUT_PULLUP | MUX_MODE0) /* spi0_d1.spi0_d1 */
+            BONE_P9_17 (PIN_OUTPUT_PULLUP | MUX_MODE0) /* spi0_cs0.spi0_cs0 */
+
+            /* GPIO pins for nRF24L01+ */
+            BONE_P9_15 (PIN_OUTPUT_PULLDOWN | MUX_MODE7)  /* gpmc_a0.gpio1_16 */
+            BONE_P9_23 (PIN_OUTPUT_PULLDOWN | MUX_MODE7)  /* gpmc_a1.gpio1_17 */
+		>;
+	};
+};
